### PR TITLE
cadl-ranch-specs, add basic test case of clientInitialization

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

**Review by chunyu3 — COMMENTED** [review]
We may need add more complex cases:

- more complex client hierarchy, e.g. more than 3 layer clients
- co-operate with @client / @operationGroup decorator
- with paramMaps to map a parameter to the property of the Client initialization Model
- a senario that not all operations has the bumped parameter

**Comment by weidongxu-microsoft** [comment]
> We may need add more complex cases:
> 
> * more complex client hierarchy, e.g. more than 3 layer clients
> * co-operate with @client / @operationGroup decorator
> * with paramMaps to map a parameter to the property of the Client initialization Model
> * a senario that not all operations has the bumped parameter

Wondering whether we need to have a subfolder for different scenarios. And this be a "basic" folder.

As the characteristic of the "client hierarchy", some of them cannot co-exist in same main.tsp. E.g. client by namespace/interface cannot be in same file as client by `@client`/`@operationGroup`.

**Comment by tadelesh** [comment]
@weidongxu-microsoft please hold this. i'm going to redesign all the client related thing in tcgc to consolidate all things related to clients, it may cause changes for decorator.

**Comment by weidongxu-microsoft** [comment]
> @weidongxu-microsoft please hold this. i'm going to redesign all the client related thing in tcgc to consolidate all things related to clients, it may cause changes for decorator.

Sure.

Note that Face API already uses this decorator in their specs (and Java emitter also uses the `.initialization` in TCGC).

File: packages/cadl-ranch-specs/http/client/initialization/basic/main.tsp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+model SubClientOptions {
+  name: string;
+}
+
+@clientInitialization(SubClientOptions)
+interface SubClient {
+  @post
+  @route("/sub-client/{name}:action")
+  action(@path name: string): void;
```
Comment (XiaofeiCao) [inline packages/cadl-ranch-specs/http/client/initialization/basic/main.tsp:30]: I'm not sure, maybe add another parameter that will be taken in method signature, as a contrast?

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/client/initialization/basic/main.tsp:30]: Sure, I will first update to our local test first, as this one likely going to take time.
https://github.com/microsoft/typespec/pull/5027

**Pull request closed without merging**